datalad_next/processors/lines_processor.py

```diff
@@ -0,0 +1,123 @@
+""" Generator the emits only complete lines """
+
+from __future__ import annotations
+
+from typing import (
+    Generator,
+    Iterable,
+)
+
+
+__all__ = ['lines_processor']
+
+
+def lines_processor(iterable: Iterable[bytes | str],
+                    separator: str | bytes | None = None,
```

Comment:
What about `separator` -> `sep`? This would align the names with `str.split()` and `str.splitlines()`

Reply:
Will do

Reply:
Thought about this again. Not a strong preference. It also makes sense to have it spelled out.

Reply:
I adopted the name from your merge PR, i.e. `itemize`.

I am not too happy with it though, it seems too generic, especially because an `itemize(iterable)` without any further parameters will yield lines.

What about one of the following?

- `iter_lines`
- `build_lines`
- `line_builder`
- `linify`  (don't take this one too seriously)

Reply:
> What about `separator` -> `sep`? This would align the names with `str.split()` and `str.splitlines()`

Done in f309fb3ebcbf670bb8ba9c00e316ba19bef68af0 (and used in 9468963e978af249ee1063e6166af261bacc6c63)

Reply:
My thinking is this: The only use case so far does not involved "lines" (zero-byte delimited items).

Actually, there is quite some complexity in the code to also mimic `splitlines()`. I considered ripping it out, but it clearly has value.

I would be OK (and even in favor of) not having a default argument for `separator`.

I am currently working on a 2nd use case for the newly merged code and it involved tab-delimited items (again no lines).

Let's remove the default argument?

Reply:
> My thinking is this: The only use case so far does not involved "lines" (zero-byte delimited items).
> 

I think that is not entirely correct. We process "proper" lines as result of `git annex examinekey --batch` output.

> [...]
>
> Let's remove the default argument?

Should we keep the option to pass `None`? I think it would make sense, because python's builtin `splitline` knows about Unix, OSX, and Windows line terminators. In addition, if operating on strings, it knows about a number of Unicode characters that terminate lines. The latter might not really be relevant, but an OS-agnostic splitlines-operation might come in handy.

Reply:
> Should we keep the option to pass `None`?

Yes, that is what I meant by "but it clearly has value"

Reply:
Let's remove the default argument. I approved the PR